### PR TITLE
gitserver: replace deprecated P4Exec command with stub unimplemented interface

### PR DESCRIPTION
**Pull request by ggilmore**
As a follow up to https://github.com/sourcegraph/sourcegraph/pull/59997, this PR replaces the implementation of the `P4Exec` RPC with a stub that simply returns `codes.Unimplemented` (see https://grpc.github.io/grpc/core/md_doc_statuscodes.html for more detail on grpc status codes). This avoids the need to refactor this usage of `p4` to use the new `perforce.NewBaseCommand` abstraction. 

This RPC endpoint is already unused by our current gitserver client. 

Any old gitserver clients that still contact this endpoint (e.g during an upgrade) will receive the Unimplemented error. However, this error shouldn't prevent a rollout - so this brief temporary state should be fine. 

## Test plan

CI

### CODE REVIEW

**Comment by ggilmore** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59999?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #59997** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59997?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #59998** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59998?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #59999** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59999?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59999?utm_source=stack-comment).

File: cmd/gitserver/internal/server_grpc.go [hunk 1]
```diff
@@ -987,3 +987,19 @@ func (gs *grpcServer) ReadFile(req *proto.ReadFileRequest, ss proto.GitserverSer
 	_, err = io.Copy(w, r)
 	return err
 }
+
+func (gs *grpcServer) P4Exec(_ *proto.P4ExecRequest, _ proto.GitserverService_P4ExecServer) error {
```
Comment (eseliger) [inline cmd/gitserver/internal/server_grpc.go:991]: idea: could just not overwrite it, and the unimplemented embedded type would throw unimplemented errors directly.

File: cmd/gitserver/internal/server.go [hunk 2]
```diff
@@ -836,6 +836,15 @@ func (s *Server) CloneRepo(ctx context.Context, repo api.RepoName, opts CloneOpt
 	return "", nil
 }
 
+func (s *Server) P4Exec(_ context.Context, logger log.Logger, _ *p4ExecRequest, _ io.Writer) execStatus {
```
Comment (eseliger) [inline cmd/gitserver/internal/server.go:839]: I don't think we need this anymore - do we? I think we could remove it.

**Review by eseliger — APPROVED** [review]
in case you won't backport this PR, it would also be okay to remove the endpoint entirely (also from the proto). Post-5.3 tasks includes to remove this entirely anyways (it's been unused for a while).

**Comment by ggilmore** [comment]
### Merge activity

* **Feb 6, 1:33 PM EST**: [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/59999) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**